docs/reports/double_descend.md

```diff
@@ -0,0 +1,127 @@
+# Report on the case of double descent in active learning
+
+Double descent was first investigated by [OpenAI in 2019](https://openai.com/blog/deep-double-descent/).
+They have also published their hypothesis on why double descent happens in some training tasks, in
+[this paper](https://arxiv.org/abs/1912.02292).
+
+During our first rounds of experiments with MCDropout for active learning, we came around double descent incident in
+several cases and hence, it was all the reason to figure out why and when it happens. Double descent is harmful to the
+active learning procedure as it is a sign that the model did not learn the current labelled set properly and therefore
+the next set of uncertainties would not be valid.
+
+A combination of several conditions and hyper-parameters could result in a double descent. Below, we will define each 
+condition/hyper-parameter, and then we describe what combinations would result in double descent occurrence. We will end
+this blog post with our hypothesis as to why double descent happens in each case.
+
+### Training Condition
+#### Early Stopping
+While early stopping helps prevent overfitting, experiments show that in the case of active
+learning early stopping along with other conditions would result in the negative effect of double descent. In our
+experiments, the trials which are tested using early stopping are marked by `lbw: Y` and `p` will define the number of
+patience epochs to make sure that the loss is not oscillating.
+
+#### Resetting the weights of the model
+During active learning, we reset the model weights to some initial/random weights after
+each active learning step, so that the model is ready to learn from the new distribution of labelled samples without
+the prior bias of the previous distribution of labelled samples. Some studies show that resetting the weights for the
+last few layers of a network could be enough for removing this prior bias and hence, it would speed up the process by
+not forcing the model to learn from scratch. However, depending on whether we are using early stopping or not this can
+introduce double descent and be harmful to the procedure. In our experiments
```

Comment:
```suggestion
the prior bias of the previous distribution of labelled samples. In this [paper](https://arxiv.org/pdf/1811.12535v1.pdf), it is shown that the last few layers have the most influential weights for estimating model uncertainty in Bayesian neural architecture, and hence, we have made it possible in our pipeline to do active learning with only resetting the weights for the
last few layers of a network. In this study, you will see if taken proper measurements resetting only the linear layers could be enough for removing this prior bias, and hence, it would speed up the process by
not forcing the model to learn from scratch. However, depending on whether we are using early stopping or not this can
introduce double descent and be harmful to the procedure. In our experiments
```